### PR TITLE
containerapp allow yaml without environment

### PR DESCRIPTION
**Pull request by Greedygre**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command

When --environment is provided and the environmentId value does not exist in --yaml, the value in --environment is used.
When the environmentId value existed in --yaml, use the environmentId in --yaml.
```
az containerapp create --yaml --environment
az containerapp job create --yaml --environment
```

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

File: src/containerapp/HISTORY.rst [hunk 1]
```diff
@@ -7,6 +7,7 @@ upcoming
 * Removed preview tag for some command groups and params (e.g. 'az containerapp job', 'az containerapp env storage', 'az containerapp env workload-profile')
 * 'az containerapp env': --enable-workload-profiles allowed values:true, false
 * 'az containerapp auth': support --token-store, --sas-url-secret, --sas-url-secret-name, --yes
+* 'az containerapp create', 'az containerapp job create': When --environment is provided and the environmentId value does not exist in --yaml, use the value in --environment as environmentId
```
Comment (Juliehzl) [inline src/containerapp/HISTORY.rst:10]: use `/` to separate

File: src/containerapp/azext_containerapp/containerapp_decorator.py [hunk 2]
```diff
@@ -605,7 +605,7 @@ def post_process(self, r):
         return r
 
     def set_up_create_containerapp_yaml(self, name, file_name):
-        if self.get_argument_image() or self.get_argument_managed_env() or self.get_argument_min_replicas() or self.get_argument_max_replicas() or self.get_argument_target_port() or self.get_argument_ingress() or \
```
Comment (Juliehzl) [inline src/containerapp/azext_containerapp/containerapp_decorator.py:608]: when there is env provided, we should provide the warning, right?

File: src/containerapp/azext_containerapp/containerapp_decorator.py [hunk 3]
```diff
@@ -660,15 +660,19 @@ def set_up_create_containerapp_yaml(self, name, file_name):
             del self.containerapp_def["workloadProfileName"]
 
         # Validate managed environment
+        env_id = self.containerapp_def["properties"]['environmentId']
+        env_info = None
+        if self.get_argument_managed_env():
+            if not self.get_argument_disable_warnings() and env_id is not None and env_id != self.get_argument_managed_env():
```
Comment (Juliehzl) [inline src/containerapp/azext_containerapp/containerapp_decorator.py:666]: it is unnecessary to re-write the warning here?

**Review by Juliehzl — APPROVED** [review]

File: src/containerapp/azext_containerapp/containerapp_decorator.py (hunk 3 above)
Comment (Greedygre) [reply src/containerapp/azext_containerapp/containerapp_decorator.py:666]: Will consider fixing this for all input args in future.

**Comment by Greedygre** [comment]
Hi @zhoxing-ms 
This PR has been reviewed by zunli, could you please help to review and merge? Thanks a lot.

**Pull request merged**